benefits/oauth/client.py

```diff
@@ -42,23 +39,20 @@ def _authorize_params(scheme):
     return params
 
 
-def register_providers(oauth_registry):
+def register_provider(oauth_registry, provider):
     """
-    Register OAuth clients into the given registry, using configuration from AuthProvider models.
+    Register OAuth clients into the given registry, using configuration from AuthProvider model.
 
     Adapted from https://stackoverflow.com/a/64174413.
     """
-    logger.info("Registering OAuth clients")
-
-    providers = AuthProvider.objects.all()
+    logger.debug(f"Registering OAuth client: {provider.client_name}")
 
-    for provider in providers:
-        logger.debug(f"Registering OAuth client: {provider.client_name}")
+    client = oauth_registry.register(
```

Comment:
At first I was wondering about this line since the [Authlib documentation](https://docs.authlib.org/en/latest/client/api.html#module-authlib.integrations.django_client) does not mention a return type for `.register()`, but after running the app in debug mode and looking at the [Authlib code](https://github.com/lepture/authlib/blob/master/authlib/integrations/base_client/registry.py#L87C9-L87C40), it turns out that `.register()` does return a client (`<authlib.integrations.django_client.apps.DjangoOAuth2App object`), which is the same client shown in `oauth._clients` after calling `register()` on `oauth`.

Reply:
> I was wondering about the difference between `oauth_client = register_provider(oauth, verifier.auth_provider)` and `oauth_client = oauth.create_client(verifier.auth_provider.client_name)`
> 
> and when to use one over the other 🤔. It looks like both of them return the same object and calling `register_provider()` multiple times is safe, so they seem to be equivalent (as long as a provider has been registered). I guess this observation may relate to Kegan's [comment](https://github.com/cal-itp/benefits/pull/2152/files#r1633957729).

Yeah, so you'd want to use `register_provider(oauth, verifier.auth_provider)` instead of `oauth.create_client(verifier.auth_provider.client_name)` when it's possible that the AuthProvider's config hasn't been registered with Authlib yet. 

I used `register_provider` in the `login` view but not in `authorize` because we know `authorize` will always be preceded by `login` and therefore the client will already be registered. But as @thekaveman is suggesting, it wouldn't hurt to do that in `authorize` too, so I'm good with making that change.

Ideally I would've liked to somehow check if the `client_name` has already been registered, call `register` for it if it hasn't, and then call `create_client`. But Authlib doesn't provide a way for us to check the registry, so all we can do is call `register` every time.

Reply:
> But Authlib doesn't provide a way for us to check the registry, so all we can do is call register every time.

What happens when we try to get a `client_name` that hasn't been registered yet? Exception, or does it just return `None`? Either way, we could handle that case and use it to register. Wrap both cases up in a helper function and call it a day :sunglasses:

Reply:
I think an exception could mean a misconfiguration, so I'll leave that case out. But I went ahead and made the helper function that registers the client if we got `None` back - [4d2e1ec](https://github.com/cal-itp/benefits/pull/2152/commits/4d2e1ec854acb45f25b9b770492becdd636fc7c4). Hopefully this code reads as more intuitive.